arroyo/processing/processor.py

```diff
@@ -85,6 +71,30 @@ def on_partitions_revoked(partitions: Sequence[Partition]) -> None:
             self.__processing_strategy = None
             self.__message = None  # avoid leaking buffered messages across assignments
 
+        def _create_strategy() -> None:
+            self.__processing_strategy = self.__processor_factory.create(self.__commit)
+            logger.debug(
+                "Initialized processing strategy: %r", self.__processing_strategy
+            )
+
+        def on_partitions_assigned(partitions: Mapping[Partition, int]) -> None:
+            logger.info("New partitions assigned: %r", partitions)
+            if self.__processing_strategy is not None:
+                _close_strategy()
+            _create_strategy()
+
+        def on_partitions_revoked(partitions: Sequence[Partition]) -> None:
+            logger.info("Partitions revoked: %r", partitions)
+            _close_strategy()
+
+            # Recreate the strategy if the consumer still has other partitions
+            # assigned and is not closed or errored
+            try:
+                if self.__consumer.tell().keys() - set(partitions):
+                    _create_strategy()
+            except Exception:
+                pass
```

Comment:
Should we crash instead? If you cannot call `tell` we would not be creating the strategy and we would run into an inconsistent state.

Reply:
Yeah, this was mostly just to catch consumer closed / terminated exception. I can just catch those specific errors instead and let anything else crash.

Reply:
Actually on second thoughts, I think I would leave this as-is. it we cannot call `tell` it only means that the consumer is in the process of shutting down and we should not be re-creating the strategy otherwise it'll mess with the shutdown sequence.

Reply:
log a warning ?

Reply:
Ended up changing this line to catch RuntimeError only. So now it will crash on anything else including networking issues.